### PR TITLE
Spring boot online book store add global exc handler

### PR DESCRIPTION
**Pull request by snowfallen**

### CODE REVIEW

File: src/main/java/book/store/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+    @Length(min = 2)
     private String title;
+    @NotNull
+    @Length(min = 2)
     private String author;
+    @NotNull
+    @Length(min = 13, max = 13)
     private String isbn;
+    @NotNull
+    @Min(1)
```
Comment (okuzan) [inline src/main/java/book/store/dto/CreateBookRequestDto.java]: @ Positive

File: src/main/java/book/store/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+    @NotNull
+    @Length(min = 2)
     private String title;
+    @NotNull
+    @Length(min = 2)
     private String author;
+    @NotNull
+    @Length(min = 13, max = 13)
     private String isbn;
+    @NotNull
```
Comment (okuzan) [inline src/main/java/book/store/dto/CreateBookRequestDto.java]: notBlank is usually what we need for strings

File: src/main/java/book/store/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
```
Comment (okuzan) [inline src/main/java/book/store/exception/CustomGlobalExceptionHandler.java]: add handler for general Exception, which will handle everything not handled by specific handlers

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,13 +1,25 @@
...
+    @NotBlank
+    @Length(min = 2)
     private String author;
+    @NotBlank
+    @Length(min = 13, max = 13)
     private String isbn;
+    @Positive
+    @NotNull
     private BigDecimal price;
     private String description;
```
Comment (Sanderfuck) [inline src/main/java/book/store/dto/CreateBookRequestDto.java:24]: You can validate description too

**Review by Sanderfuck — APPROVED** [review]

File: src/main/java/book/store/exception/GlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<List<String>> handleValidationExceptions(
+            MethodArgumentNotValidException ex) {
+
```
Comment (MentorMateAcademy) [inline src/main/java/book/store/exception/GlobalExceptionHandler.java]: handleMethodArgumentNotValid where does this method?

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**